ui/jobs/bars.ts

```diff
@@ -0,0 +1,784 @@
+    // GP Alarm
+    if (data.gp < this.options.GpAlarmPoint) {
+      return true;
+    } else if (data.gpAlarmReady && !data.gpPotion && data.gp >= this.options.GpAlarmPoint) {
+      const audio = new Audio('../../resources/sounds/freesound/power_up.webm');
+      audio.volume = this.options.GpAlarmSoundVolume;
+      void audio.play();
+      return false;
+    }
+    return true;
+  }
+
+  _updateOpacity(transparent: boolean): void {
+    const opacityContainer = document.getElementById('opacity-container');
+    if (!opacityContainer)
+      return;
+    opacityContainer.style.opacity = transparent
+      ? this.options.OpacityOutOfCombat.toString()
+      : '1.0';
+  }
+
+  _updateFoodBuff(o: {
+    inCombat: boolean;
+    contentType?: number;
+    foodBuffExpiresTimeMs: number;
+    foodBuffTimer: number;
+  }): number | undefined {
+    // Non-combat jobs don't set up the left buffs list.
+    if (!this.o.leftBuffsList)
+      return;
+
+    const CanShowWellFedWarning = () => {
+      if (!this.options.HideWellFedAboveSeconds)
+        return false;
+      if (o.inCombat)
+        return false;
+      if (o.contentType === undefined)
+        return false;
+      return kWellFedContentTypes.includes(o.contentType);
+    };
+
+    // Returns the number of ms until it should be shown. If <= 0, show it.
+    const TimeToShowWellFedWarning = () => {
+      const nowMs = Date.now();
+      const showAtMs = o.foodBuffExpiresTimeMs - (this.options.HideWellFedAboveSeconds * 1000);
+      return showAtMs - nowMs;
+    };
+
+    window.clearTimeout(o.foodBuffTimer);
+    o.foodBuffTimer = 0;
+
+    const canShow = CanShowWellFedWarning();
+    const showAfterMs = TimeToShowWellFedWarning();
+
+    if (!canShow || showAfterMs > 0) {
+      this.o.leftBuffsList.removeElement('foodbuff');
+      if (canShow)
+        return window.setTimeout(this._updateFoodBuff.bind(this), showAfterMs);
+    } else {
+      const div = makeAuraTimerIcon(
+        'foodbuff',
+        -1,
+        1,
+        this.options.BigBuffIconWidth,
+        this.options.BigBuffIconHeight,
+        '',
+        this.options.BigBuffBarHeight,
+        this.options.BigBuffTextHeight,
+        'white',
+        this.options.BigBuffBorderSize,
+        'yellow',
+        'yellow',
+        foodImage,
+      );
+      this.o.leftBuffsList.addElement('foodbuff', div, -1);
+    }
+  }
+
+  _setPullCountdown(seconds: number): void {
+    if (!this.o.pullCountdown)
+      return;
+
+    const inCountdown = seconds > 0;
+    const showingCountdown = this.o.pullCountdown.duration ?? 0 > 0;
+    if (inCountdown !== showingCountdown) {
+      this.o.pullCountdown.duration = seconds;
+      if (inCountdown && this.options.PlayCountdownSound) {
+        const audio = new Audio('../../resources/sounds/freesound/sonar.webm');
+        audio.volume = 0.3;
+        void audio.play();
+      }
+    }
+  }
+
+  setJobsContainerVisibility(hide?: boolean): void {
```

Comment:
If the parameter is hide, maybe this should be called `setJobsContainerHidden` to avoid confusion? (or maybe it should be `setJobsContainerVisible(show?: boolean)`?

Reply:
Hmm, I think to change it as `show` would be better.